core/utils/bytes_utils.hpp

```diff
@@ -79,6 +79,13 @@ struct bytes_io<T, sizeof(uint16_t)> {
     return out;
   }
 
+  static T read(byte_type*& in) {
+    const T out = ((in[0] << 8) | in[1]);
```

Comment:
sizeof(T) is expected to be always 2? Otherwise we will read into T 2 bytes but advance by actual size of T which seems odd

Reply:
Ah indeed its for uint16_t